java.lang.NoSuchMethodError

Hello all,

while testing a new version of rundeck-cli I fould that v2.0.6 and 2.0.7 give an error while validating an ACL policy. 
I am using only [this official ACL policy](https://docs.rundeck.com/docs/manual/document-format-reference/aclpolicy-v10.html#example-admin-policy)
```
root@rdk-molecule:/# /var/lib/rundeck-cli/rd-cli-tool-shadow-2.0.7/bin/rd acl validate -d /etc/rundeck
picocli.CommandLine$ExecutionException: Error while calling command (public int org.rundeck.client.ext.acl.Acl.validate(org.rundeck.client.ext.acl.Acl$AclOptions)): java.lang.NoSuchMethodError: 'void org.yaml.snakeyaml.constructor.Constructor.<init>(java.lang.Class)'
        at picocli.CommandLine.executeUserObject(CommandLine.java:1986)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at org.rundeck.client.tool.Main.main(Main.java:155)
Caused by: java.lang.NoSuchMethodError: 'void org.yaml.snakeyaml.constructor.Constructor.<init>(java.lang.Class)'
        at com.dtolabs.rundeck.core.authorization.providers.yaml.model.YamlPolicyDocConstructor.<init>(YamlPolicyDocConstructor.java:31)
        at com.dtolabs.rundeck.core.authorization.providers.YamlParsePolicy$4.loadAll(YamlParsePolicy.java:743)
        at com.dtolabs.rundeck.core.authorization.providers.YamlPolicyCollection.load(YamlPolicyCollection.java:84)
        at com.dtolabs.rundeck.core.authorization.providers.YamlPolicyCollection.<init>(YamlPolicyCollection.java:54)
        at com.dtolabs.rundeck.core.authorization.providers.YamlPolicyFactoryV2.policiesFromSource(YamlPolicyFactoryV2.java:39)
        at com.dtolabs.rundeck.core.authorization.providers.PolicyCollectionFactory.validate(PolicyCollectionFactory.java:68)
        at com.dtolabs.rundeck.core.authorization.providers.YamlProvider.validate(YamlProvider.java:98)
        at com.dtolabs.rundeck.core.authorization.providers.YamlProvider.validate(YamlProvider.java:85)
        at com.dtolabs.rundeck.core.authorization.providers.YamlProvider.validate(YamlProvider.java:89)
        at org.rundeck.client.ext.acl.Acl.validatePolicies(Acl.java:1080)
        at org.rundeck.client.ext.acl.Acl.validate(Acl.java:584)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1972)
        ... 8 more
```
OS: Ubuntu:2204
Rundeck version: 4.17.1
rundeck-cli: 2.0.6 & 2.0.7

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -18,7 +18,7 @@ jacksonDatabind = "2.15.3"
 picocli = "4.6.3"
 snakeYaml = "2.0"
 #used for authz lib integration
-rundeck = "4.17.2-20231107"
+rundeck = "4.17.3-20231113"
 testcontainers = "1.17.2"
 
 [libraries]

